tests/scripts/quiet/quiet.sh

```diff
@@ -22,9 +22,16 @@
 #              be silenced, e.g. " --version | test ". In this example, "make lib test" will
 #              not be silent, but "make lib" will be.
 
-# Locate original tool
-TOOL_WITH_PATH=$(dirname "$0")/$TOOL
-ORIGINAL_TOOL=$(type -ap "${TOOL}" | grep -v -Fx "$TOOL_WITH_PATH" | head -n1)
+# Function to normalise paths
+get_real_filename() {
+    leafname="$(basename "$1")"
+    ( cd $(dirname "$1"); echo "$(pwd)/$leafname" )
+}
+
+# Normalise path to wrapper script
+WRAPPER_WITH_PATH=$(get_real_filename "$0")
+# Identify original tool (compare normalised path to WRAPPER_WITH_PATH to avoid recursively calling the same wrapper script)
+ORIGINAL_TOOL="$(for p in $(type -ap "$TOOL"); do get_real_filename "$p" ; done | grep -v -Fx "$WRAPPER_WITH_PATH" | head -n1)"
```

Comment:
This code is rather complicated. You can use the `-ef` test operator to check if two paths point to the same file (not POSIX, but present in bash as well as almost all shells you'll encounter these days), avoiding the complex file name manipulations and following file symlinks (which isn't really a concern here).

`for p in $(type -ap "$TOOL")` breaks with paths containing spaces. Parsing a newline-delimited list in POSIX sh is unfortunately fiddly. (Bash ≥4 has `mapfile` and zsh has `"${(@f)$(…)}"` but that doesn't help with macOS's `/bin/bash` which is 3.2.) One solution is to disable globbing and set `IFS` to newlines only. Another way is `IFS= read -r line`. Note that `somecommand | read … line; … $line` doesn't work because the right-hand side of the pipe operator runs in a subshell so the `line` variable is not set in the following command. But you can use `read … <(somecommand)` as long as you don't need to catch failures of `somecommand`.

```
while IFS= read -r ORIGINAL_TOOL < <(type -ap -- "$TOOL"); do
    if ! [[ $ORIGINAL_TOOL -ef $0 ]]; then break; fi
done
```

Reply:
Thanks, this is cleaner. But AIUI the redirection needs to happen after the `done`?

Reply:
Oh, yes, sorry. The redirection where I put it only works in the common case where `type` only emits one line.